src/EditorFeatures/Core/Implementation/TodoComment/TodoCommentIncrementalAnalyzer.cs

```diff
@@ -56,38 +57,46 @@ public async Task AnalyzeSyntaxAsync(Document document, InvocationReasons reason
                 return;
             }
 
+            var (existingData, newData) = await ComputeAndPersistTodoCommentsAsync(document, _state, _todoCommentTokens, cancellationToken).ConfigureAwait(false);
+
+            // * NOTE * cancellation can't throw after this point.
```

Comment:
this comment could use more details.  i can probaby infer it, but it might be worth stating something explicit like "we have persisted and updated our new view of the todo commetns in memory, if we don't update the UI, we can now be in an inconsistent state where our view and what we're showing the user are not in sync."

Reply:
Ah, I don't have much context either on the original cancellation logic - I am just moving existing code around.